Sources/Purchasing/OfferingsManager.swift

```diff
@@ -275,6 +275,95 @@ private extension OfferingsManager {
         }
     }
 
+    private func fetchProducts(
+        withIdentifiers identifiers: Set<String>,
+        fromResponse response: OfferingsResponse,
+        completion: @escaping ProductsManagerType.Completion
+    ) {
+        if self.systemInfo.dangerousSettings.uiPreviewMode {
+            let mockProducts = self.createMockProducts(productIdentifiers: identifiers, fromResponse: response)
+            completion(.success(mockProducts))
+        } else {
+            self.productsManager.products(withIdentifiers: identifiers, completion: completion)
+        }
+    }
+
+    // MARK: - Mocks for UI Preview mode
+
+    private func createMockProducts(
+        productIdentifiers: Set<String>,
+        fromResponse response: OfferingsResponse
+    ) -> Set<StoreProduct> {
+        let packagesByProductID = response.packages.dictionaryAllowingDuplicateKeys { $0.platformProductIdentifier }
+        let products = productIdentifiers.map { identifier -> StoreProduct in
+            let productType = self.mockProductType(from: packagesByProductID[identifier],
+                                                   productIdentifier: identifier)
+
+            let introductoryDiscount: TestStoreProductDiscount? = {
+                // For mocking purposes: all yearly subscriptions have a 1-week free trial
+                guard productType.period?.unit == .year else { return nil }
+                return TestStoreProductDiscount(
+                    identifier: "intro",
+                    price: 0,
+                    localizedPriceString: "$0.00",
+                    paymentMode: .freeTrial,
+                    subscriptionPeriod: SubscriptionPeriod(value: 1, unit: .week),
+                    numberOfPeriods: 1,
+                    type: .introductory
+                )
```

Comment:
Based on @JayShortway's comment [here](https://github.com/RevenueCat/purchases-ios/pull/4693#discussion_r1924834946), I'm only adding mocked introductory offers for yearly subscriptions